xcell/tests/test_data.py

```diff
@@ -0,0 +1,237 @@
+import glob
+import pytest
+import os
+import yaml
+from xcell.cls.data import Data
+from ..mappers import mapper_from_name
+
+
+def read_yaml_file(file):
+    with open(file) as f:
+        config = yaml.safe_load(f)
+    return config
+
+
+def get_input_file():
+    return './xcell/tests/data/desy1_ebossqso_p18cmbk.yml'
+
+
+def get_config_dict():
+    input_file = get_input_file()
+    config = read_yaml_file(input_file)
+    return config
+
+
+def get_data():
+    input_file = get_input_file()
+    return Data(input_file)
+
+
+def get_data_from_dict():
+    input_dict = get_config_dict()
+    return Data(data=input_dict)
+
+
+def remove_yml_file(config):
+    outdir = config['output']
+    fname = os.path.join(outdir, '*.yml')
+    for f in glob.glob(fname):
+        os.remove(f)
+
+
+def remove_outdir(config):
+    outdir = config['output']
+    if outdir != './xcell/tests/cls/':
+        raise ValueError('Outdir looks odd. Not removing it as precaution')
+    # Using os.rmdir as it will only remove an empty directory
+    elif os.path.isdir(outdir):
+        os.rmdir(outdir)
+
+
+def test_initizalization():
+    input_file = get_input_file()
+    config = read_yaml_file(input_file)
+    with pytest.raises(ValueError):
+        Data(input_file, data=config)
+
+    config['bpw_edges'] = [0, 10]
+    data2 = Data(data=config, override=True)
+    assert os.path.isfile('xcell/tests/cls/data.yml')
+    assert data2.data['bpw_edges'] == [0, 10]
+    remove_yml_file(config)
+
+
+def test_read_data():
+    data = get_data()
+    input_file = get_input_file()
+    config = get_config_dict()
+    assert data.read_data(input_file) == config
+
+    remove_yml_file(config)
+
+
+def test_read_saved_data():
+    data = get_data()
+    data2 = get_data()
+    assert data.data == data2.data
```

Comment:
Humm, this test will pass even if it's not reading from the saved yaml file, right?

Reply:
Fixed